controllers/scopetemplate_controller_test.go

```diff
@@ -0,0 +1,548 @@
+						err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
+							Namespace: namespace2.GetName(),
+						},
+							client.MatchingLabels{
+								scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
+								scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+								clusterRoleBindingGenerateKey: "test",
+							})
+						if err != nil {
+							return err
+						}
+
+						if len(roleBindingList.Items) != 1 {
+							return fmt.Errorf("Expected 1 roleBinding, found %d", len(roleBindingList.Items))
+						}
+
+						return nil
+					}, timeout, interval).Should(BeNil())
+
+					existingRB := &roleBindingList.Items[0]
+
+					Expect(len(existingRB.OwnerReferences)).To(Equal(1))
+					Expect(existingRB.OwnerReferences).Should(ContainElement(metav1.OwnerReference{
+						APIVersion: "operators.io.operator-framework/v1",
+						Kind:       "ScopeInstance",
+						Name:       scopeInstance.GetObjectMeta().GetName(),
+						UID:        scopeInstance.GetObjectMeta().GetUID(),
+					}))
+
+					Expect(len(existingRB.Subjects)).To(Equal(1))
+					Expect(existingRB.Subjects).Should(ContainElement(rbacv1.Subject{
+						Kind:     "Group",
+						Name:     "manager",
+						APIGroup: "rbac.authorization.k8s.io",
+					}))
+					Expect(existingRB.RoleRef).To(Equal(rbacv1.RoleRef{
+						Kind:     "ClusterRole",
+						Name:     "test",
+						APIGroup: "rbac.authorization.k8s.io",
+					}))
+
+					roleBindingList = &rbacv1.RoleBindingList{}
+					Eventually(func() error {
+						err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
+							Namespace: namespace.GetName(),
+						},
+							client.MatchingLabels{
+								scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
+								scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+								clusterRoleBindingGenerateKey: "test",
+							})
+						if err != nil {
+							return err
+						}
+
+						if len(roleBindingList.Items) != 1 {
+							return fmt.Errorf("Expected 1 roleBinding, found %d", len(roleBindingList.Items))
+						}
+
+						return nil
+					}, timeout, interval).Should(BeNil())
+
+					existingRB = &roleBindingList.Items[0]
+
+					Expect(len(existingRB.OwnerReferences)).To(Equal(1))
+					Expect(existingRB.OwnerReferences).Should(ContainElement(metav1.OwnerReference{
+						APIVersion: "operators.io.operator-framework/v1",
+						Kind:       "ScopeInstance",
+						Name:       scopeInstance.GetObjectMeta().GetName(),
+						UID:        scopeInstance.GetObjectMeta().GetUID(),
+					}))
+
+					Expect(len(existingRB.Subjects)).To(Equal(1))
+					Expect(existingRB.Subjects).Should(ContainElement(rbacv1.Subject{
+						Kind:     "Group",
+						Name:     "manager",
+						APIGroup: "rbac.authorization.k8s.io",
+					}))
+					Expect(existingRB.RoleRef).To(Equal(rbacv1.RoleRef{
+						Kind:     "ClusterRole",
+						Name:     "test",
+						APIGroup: "rbac.authorization.k8s.io",
+					}))
+				})
```

Comment:
This could probably moved to it's own function, parameters could include the roleBinding, scopeTemplate, and scopeInstance. The roleBinding's values can then be compared against the expected values defined by the scopeTemplate, and scopeInstance.

Reply:
created separate functions and reduced code for this.